snuba/query/extensions.py

```diff
@@ -39,3 +45,19 @@
     'required': ['project'],
     'additionalProperties': False,
 }
+
+
+def get_time_limit(timeseries_extension: Mapping[str, Any]) -> Tuple[int, int]:
+    max_days, date_align = state.get_configs([
+        ('max_days', None),
+        ('date_align_seconds', 1),
+    ])
+
+    to_date = util.parse_datetime(timeseries_extension['to_date'], date_align)
+    from_date = util.parse_datetime(timeseries_extension['from_date'], date_align)
+    assert from_date <= to_date
+
+    if max_days is not None and (to_date - from_date).days > max_days:
+        from_date = to_date - timedelta(days=max_days)
```

Comment:
I never really thought about this, but it bugs me a little that we're doing this silently. It might make sense for some later version of the response format to include an "warnings" data structure where things like this would be recorded so at least there's some visibility into how the query is being adjusted.